File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    const sum1 = queue1.reduce((acc, cur) => acc + cur, 0);
    const sum2 = queue2.reduce((acc, cur) => acc + cur, 0);
    const totalSum = sum1 + sum2;
    if (totalSum % 2 !== 0) return -1;
    
    const queue = [...queue1, ...queue2];
    const targetSum = totalSum / 2;
    let currentSum = sum1;
    let left = 0;
    let right = queue1.length;
    const maxMoves = queue1.length * 3;
    
    for(let moves = 0; moves < maxMoves; moves++) {
        if (currentSum === targetSum) return moves;
        
        if (currentSum > targetSum) currentSum -= queue[left++];
        else currentSum += queue[right++ % queue.length];
    }
    
    return -1;
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
def solution(queue1, queue2):
    sum1, sum2 = sum(queue1), sum(queue2)
    if (sum1 + sum2) % 2 != 0:
        return -1
    target = (sum1 + sum2) // 2
    queue = queue1 + queue2
    
    cur_sum = sum1
    left, right = 0, len(queue1)
    max_moves = len(queue1) * 3
    
    for moves in range(max_moves): 
        if cur_sum == target:
            return moves

        if cur_sum > target:
            cur_sum -= queue[left]
            left += 1
        else:
            cur_sum += queue[right % len(queue)]
            right += 1
    return -1